firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/DataTransportMlStatsSender.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.ml.modeldownloader.internal;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import com.google.android.datatransport.Encoding;
+import com.google.android.datatransport.Event;
+import com.google.android.datatransport.Transport;
+import com.google.android.datatransport.cct.CCTDestination;
+import com.google.android.datatransport.runtime.TransportRuntime;
+
+/**
+ * This class is responsible for sending FirebaseMl Stats to Firebase through Google DataTransport.
+ *
+ * <p>These will be equivalent to FirebaseMlLogEvent.proto internally.
+ *
+ * @hide
+ */
+public class DataTransportMlStatsSender {
+  private static final String FIREBASE_ML_STATS_NAME = "FIREBASE_ML_LOG_SDK";
+  private final Transport<FirebaseMlStat> transport;
+
+  @NonNull
+  public static DataTransportMlStatsSender create(@NonNull Context context) {
+    TransportRuntime.initialize(context);
+    final Transport<FirebaseMlStat> transport =
+        TransportRuntime.getInstance()
```

Comment:
You should not be calling into the runtime, instead you should inject the `TransportFactory` instance through components(which I think you already did above), why does it need to be done here as well?

Reply:
Updated - hopefully correctly. Still crashes in my test app, but I might not be adding it right. Let me know if you have time to help.